react/features/connection-indicator/statsEmitter.js

```diff
@@ -0,0 +1,147 @@
+
+        conference.on(connectionQuality.REMOTE_STATS_UPDATED,
+            (id, stats) => this._emitStatsUpdate(id, stats));
+    },
+
+    /**
+     * Add a subscriber to be notified when stats are updated for a specified
+     * user id.
+     *
+     * @param {string} id - The user id whose stats updates are of interest.
+     * @param {Function} callback - The function to invoke when stats for the
+     * user have been updated.
+     * @returns {void}
+     */
+    subscribeToClientStats(id, callback) {
+        if (!id) {
+            return;
+        }
+
+        if (!subscribers[id]) {
+            subscribers[id] = [];
+        }
+
+        subscribers[id].push(callback);
+    },
+
+    /**
+     * Remove a subscriber that is listening for stats updates for a specified
+     * user id.
+     *
+     * @param {string} id - The user id whose stats updates are no longer of
+     * interest.
+     * @param {Function} callback - The function that is currently subscribed to
+     * stat updates for the specified user id.
+     * @returns {void}
+     */
+    unsubscribeToClientStats(id, callback) {
+        if (!subscribers[id]) {
+            return;
+        }
+
+        const filteredSubscribers = subscribers[id].filter(
+            subscriber => subscriber !== callback);
+
+        if (filteredSubscribers.length) {
+            subscribers[id] = filteredSubscribers;
+        } else {
+            delete subscribers[id];
+        }
+    },
+
+    /**
+     * Emit a stat update to all those listening for a specific user's
+     * connection stats.
+     *
+     * @param {string} id - The user id the stats are associated with.
+     * @param {Object} stats - New connection stats for the user.
+     * @returns {void}
+     */
+    _emitStatsUpdate(id, stats = {}) {
+        const callbacks = subscribers[id] || [];
+
+        callbacks.forEach(callback => {
+            callback(stats);
+        });
+    },
+
+    /**
+     * Emit a stat update to all those listening for local stat updates. Will
+     * also update listeners of remote user stats of changes related to their
+     * stats.
+     *
+     * @param {Object} stats - Connection stats for the local user as provided
+     * by the library.
+     * @returns {void}
+     */
+    _onStatsUpdated(stats) {
+        const allUserFramerates = stats.framerate;
+        const allUserResolutions = stats.resolution;
+
+        const currentUserId = APP.conference.getMyUserId();
+        const currentUserFramerate = allUserFramerates[currentUserId];
+        const currentUserResolution = allUserResolutions[currentUserId];
+
+        // FIXME resolution and framerate are hashes keyed off of user ids with
+        // stat values. Receivers of stats expect resolution and framerate to
+        // be primatives, not hashes, so overwrites the 'lib-jitsi-meet' stats
+        // objects.
+        stats.framerate = currentUserFramerate;
+        stats.resolution = currentUserResolution;
```

Comment:
wondering if it would be better to have a single object that looked like:
```
{
  id:
    {
      frameRate: XXX,
      resolution: YYY
    }
}
```

then you could only do a single loop (instead of one per stat type) and maybe even just emit this whole nested stats object (instead of emitting one stat at a time)